Objects/unicodeobject.c

```diff
@@ -8208,14 +8208,14 @@ struct encoding_map {
     PyObject_HEAD
     unsigned char level1[32];
     int count2, count3;
-    unsigned char level23[1];
+    unsigned char level23[];
```

Comment:
@gpshead: Can you please write a PR which only contains this change? Since it doesn't touch the public C API, C++ compatibility is not an issue (I don't think that Python can be built with a C++ compiler). We can start with this change and see how it goes? See also the discussion at https://github.com/python/peps/pull/1349